x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/analyzer/VerifierErrorMessagesTests.java

```diff
@@ -1257,4 +1258,62 @@ public void testBinaryFieldWithNoDocValues() {
         assertEquals("1:34: Binary field [binary] cannot be used for ordering unless it has the doc_values setting enabled",
             error("SELECT binary FROM test ORDER BY binary"));
     }
+
+    public void testDistinctIsNotSupported() {
+        assertEquals("1:8: SELECT DISTINCT is not yet supported", error("SELECT DISTINCT int FROM test"));
+    }
+
+    public void testExistsIsNotSupported() {
+        assertEquals("1:33: EXISTS is not yet supported", error("SELECT test.int FROM test WHERE EXISTS (SELECT 1)"));
+    }
+
+    public void testScoreIsNotInHaving() {
+        assertEquals("1:54: HAVING filter is unsupported for function [SCORE()]\n" +
+                "line 1:54: [SCORE()] cannot be an argument to a function",
```

Comment:
This message is rather confusing... SCORE() is not in a function in the query...

Reply:
The check at 

https://github.com/elastic/elasticsearch/blob/f539e774dc10cc004a6f637dc310b91c359c8391/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java#L707

checks that the `Score` instance is not child of any Function expression. In the test case the `GreaterThan` is also a function, so technically the error message was correct saying that the `SCORE()` function cannot be passed into other function calls (`>` in this case). Changed the error message on this check to `[SCORE()] cannot be used in expressions, does not support further processing`, I think that explains the problem better.